src/modules/claim/claim.processor.ts

```diff
@@ -1,23 +1,103 @@
-import { OnQueueError, Process, Processor } from '@nestjs/bull';
-import { Job } from 'bull';
+import {
+  InjectQueue,
+  OnQueueError,
+  OnQueueFailed,
+  Process,
+  Processor,
+} from '@nestjs/bull';
+import { Job, Queue } from 'bull';
 import { Logger } from '../logger/logger.service';
 import { ClaimService } from './claim.service';
-
+import {
+  ClaimRequestType,
+  IClaimIssuance,
+  IClaimRejection,
+  IClaimRequest,
+  NATS_EXCHANGE_TOPIC,
+} from './claim.types';
+import { NatsService } from '../nats/nats.service';
+import { Keys } from '@ew-did-registry/keys';
+import { JWT } from '@ew-did-registry/jwt';
+import { ConfigService } from '@nestjs/config';
 @Processor('claims')
 export class ClaimProcessor {
+  natsSubjectSuffix: string;
   constructor(
     private readonly claimService: ClaimService,
+    private readonly nats: NatsService,
     private readonly logger: Logger,
-  ) {}
+    private readonly config: ConfigService,
+    @InjectQueue('claims') private readonly claimQueue: Queue<string>,
+  ) {
+    this.natsSubjectSuffix = this.config.get<string>('NATS_SUBJECT_SUFFIX');
+  }
+
+  async enqueueClaimEnrollmentRequest(data: IClaimRequest) {
+    return this.claimQueue.add(
+      ClaimRequestType.CLAIM_ENROLLMENT_REQUEST,
+      JSON.stringify(data),
+    );
+  }
+
+  @Process(ClaimRequestType.CLAIM_ENROLLMENT_REQUEST)
+  public async processClaimEnrollmentRequest(job: Job<string>) {
+    const rq = JSON.parse(job.data) as IClaimRequest;
+    await this.claimService.handleClaimEnrollmentRequest(rq);
+
+    this.notifyDIDs(
+      ClaimRequestType.CLAIM_ENROLLMENT_REQUEST,
+      rq.id,
+      rq.claimIssuer,
+    );
+  }
```

Comment:
@Passerino I was thinking about this a bit more. Is `claimService.handleClaimEnrolmentRequest` idempotent? That is could we run it twice without issue?
I'm imagining a situation where `this.claimService.handleClaimEnrolmentRequest` succeeds but then `this.notifyDIDs` fails (for example because the NATS server is temporarily down). Could we run the queue job again without issue?

Reply:
whole job can be repeated according to configuration. But it anything fails , whole job will be repeated if we wish.

Reply:
If we are worried about NATS maybe we should have internal queue there in natsService...

Reply:
```suggestion
  @Process(ClaimRequestType.CLAIM_REQUEST_NOTIFICATION_JOB)
  public async processClaimEnrollmentRequest(job: Job<string>) {
    const rq = JSON.parse(job.data) as IClaimRequest;
    await this.claimService.handleClaimEnrollmentRequest(rq); -> move this back to HTTP

    this.notifyDIDs(
      ClaimRequestType.CLAIM_ENROLLMENT_REQUEST,
      rq.id,
      rq.claimIssuer,
    );
  }
```